#!/bin/bash

#SBATCH --job-name=01_aTRAM_COI_unmappedPROW953R1_20220901
#SBATCH --partition=pcon06
#SBATCH --output=01_aTRAM_COI_unmappedPROW953R1_20220901_%j.txt
#SBATCH --error=01_aTRAM_COI_unmappedPROW953R1_20220901_%j.err
#SBATCH --mail-type=ALL
#SBATCH --mail-user=alix.matthews@smail.astate.edu  
#SBATCH --nodes=1
#SBATCH --ntasks-per-node=32
#SBATCH --time=06:00:00

module load exonerate/2.4.0
module load python/3.7.3-anaconda
source /share/apps/bin/conda-3.7.3.sh
conda activate aTRAM

cd /local_scratch/$SLURM_JOB_ID/

############################################################
############ CONFIGURATION INFORMATION ############
############ LOCATIONS AND SAMPLE NAMES ############
############################################################

## SAMPLE FILE NAMES 
# this file has only the weird sample (PROW 953 R1)
filenames="/scrfs/storage/amatthews/20210816_projects/20210816_exp/20210816_exp_filenames_PROW953_only.txt"

## MAIN DIR
PROJECT_DIR=/scrfs/storage/amatthews/20210816_projects/20210816_exp

## ATRAM DIR
## mkdir beforehand
## contains COI reference (coi_Proctophyllodes_miliariae) .fasta file
ATRAM_DIR=$PROJECT_DIR/01_aTRAM_COI_20220901

## ATRAM DATABASES
ATRAM_DB=$ATRAM_DIR/aTRAM_db
mkdir ATRAM_DB


############ SET UP PER SAMPLE DIRS ############

while IFS=" " read -r value1
do {


## FASTQ FILE DIRECTORY
SAMPLE_FASTQ=$PROJECT_DIR/00_PP_20210903/Adapter_Removed_bb/${value1}

## ATRAM_DB DIR
## this next line actually doesn't do anything meaningful (forgot to put a $ before {}), but keeping because the rest of the script works and don't want removing this to mess it up!

mkdir $ATRAM_DB/{value1}

## SAMPLE OUTPUT / coi / (ALL, FILTERED, STITCH)
SAMPLE_OUT=$ATRAM_DIR/${value1}_out
mkdir $SAMPLE_OUT
SAMPLE_OUT_coi=$SAMPLE_OUT/${value1}_coi
mkdir -p $SAMPLE_OUT_coi
mkdir $SAMPLE_OUT_coi/all
mkdir $SAMPLE_OUT_coi/filtered
mkdir $SAMPLE_OUT_coi/stitch



############################################################
############ RUN aTRAM ############
############################################################

############ PREPROCESSOR ############

### RUNNING ON THE UNMAPPED (did not map to bacteria) READS!

atram_preprocessor.py --blast-db=$ATRAM_DB/${value1}/${value1} --end-1=$SAMPLE_FASTQ/${value1}_dd_bb_trim_unmapped_R1.fastq  --end-2=$SAMPLE_FASTQ/${value1}_dd_bb_trim_unmapped_R2.fastq
## second /${value1} on --blast-db line is referring to the prefix to use for the file names it creates in this step


############ ASSEMBLER ############
############ coi ############

atram.py --blast-db=$ATRAM_DB/${value1}/${value1} --query-split=$ATRAM_DIR/COI_ref.fasta --assembler=spades --protein --db-gencode=5 --output-prefix=$SAMPLE_OUT_coi/all/
## second /${value1} on --blast-db line is referring to the prefix to use/search for the file names it created in the previous step

cd $SAMPLE_OUT_coi
find . -name "*filtered_contigs.fasta" -exec mv -i {} -t "./filtered/" ";"
## move filtered_contigs files to a new folder called "filtered" (separate them from non-filtered)



############ STITCHER ############
############ coi ############

atram_stitcher.py --assemblies-dir=$SAMPLE_OUT_coi/filtered --reference-genes=$ATRAM_DIR/COI_ref.fasta --taxa=/scrfs/storage/amatthews/20210816_projects/20210816_exp/20210816_exp_filenames_PROW953_only.txt --output-prefix=$SAMPLE_OUT_coi/stitch/${value1}


############ FINISH aTRAM ############

} done <"$filenames"



